Cargo.toml

```diff
@@ -33,7 +36,12 @@ env_logger = "0.3"
 futures-cpupool = "0.1"
 clap = "2.0"
 
+[target.'cfg(target_os = "macos")'.dev-dependencies]
```

Comment:
This line will make this dependency apply only for osx. For linux, it internally uses the openssl crate, but we can make required data for testing purposes without needing to bring in the openssl crate.

For windows, I didn't write the test case since I don't actually have access to a windows machine and it seemed a little complicated based on the tests I read in the `native-tls` and `tokio-tls` crates.